components/input/Input.js

```diff
@@ -73,42 +83,91 @@ export default class Input extends Component {
   }
 
   handleIncreaseValue() {
-    this.setState((previousState, props) => ({
-      value: previousState.value ? previousState.value + props.step : props.step,
-    }));
+    this.step(1);
   }
 
   handleDecreaseValue() {
-    this.setState((previousState, props) => ({
-      value: previousState.value ? previousState.value - props.step : -props.step,
-    }));
+    this.step(-1);
+  }
+
+  hasError() {
+    const { meta } = this.props;
+    return Boolean(meta && meta.error && meta.touched);
+  }
+
+  format(number) {
+    const { precision } = this.props;
+
+    let formattedNumber = this.toNumber(number);
+
+    if (precision !== null) {
+      formattedNumber = number.toFixed(precision);
+    }
+
+    formattedNumber += '';
+
+    return formattedNumber;
```

Comment:
Maybe renaming this to something like `formatNumber` so it's easier to understand what it's actually doing? Also, why not be explicit when returning a string and actually `return String(formattedNumber)`?